src/content/docs/en/reference/loader-reference.mdx

```diff
@@ -3,6 +3,324 @@ title: Astro Loader API
+
+#### `get`
+**Type**: `(key: string) => DataEntry | undefined`
+
+Get an entry from the store by its ID. Returns `undefined` if the entry does not exist. 
+
+```ts
+const existingEntry = store.get("my-entry");
+```
+
+The returned object has the following properties:
+```ts
+interface DataEntry<TData extends Record<string, unknown>> {
+	/** The ID of the entry. Unique per collection. */
+	id: string;
+	/** The parsed entry data */
+	data: TData;
+	/** The file path of the content, if applicable. Relative to the site root. */
+	filePath?: string;
+	/** The raw body of the content, if applicable. */
+	body?: string;
+	/** An optional content digest, to check if the content has changed. */
+	digest?: number | string;
+	/** The rendered content of the entry, if applicable. */
+	rendered?: RenderedContent;
+}
+```
+
+#### `set`
+
+**Type**: `(opts: Opts) => boolean`
+
+The main method to update the store with new entries. It takes an object with the following properties:
+
+- `id: string`: The ID of the entry. Must be unique per collection. If the ID already exists in the store, the entry will be replaced.
+- `data`: The data to store. This will be stored unchanged, so if it needs to be parsed or validated then it should be done before calling `set()`.
+- `body?: string`: The raw body of the content, if applicable. This is mostly relevant for file-based loaders, or those that also store rendered content.
+- `filePath?: string`: The file path of the content, if applicable. This only applies to file-based loaders, and is used to resolve paths such as images or other assets.
+- `digest?: string`: An optional content digest, to check if the content has changed. It is an opaque value, so you can use any method to generate it, but we recommended using the `generateDigest()` function. If provided it will be checked against the existing entry, and if it matches then the entry will not be updated and `set` will return false.
+- `rendered?: { html?: string, metadata?: Record<string, unknown> }`: The rendered HTML content of the entry, if applicable. If provided, users will be able to use the `<Content />` component to render the HTML. The metadata object is used for things like frontmatter or other metadata extracted from the content.
+
+Returns `true` if the entry was set. If `digest` was passed and matched an existing entry with the same ID, it will return `false` and not update the entry.
+
+```ts title=loader.ts {7-14}
+    for (const item of feed.items) {
+      const data = await parseData({
+        id: item.guid,
+        data: item,
+      });
+      const digest = generateDigest(data);
+      store.set({
+        id,
+        data,
+        rendered: {
+          html: data.description ?? "",
+        },
+        digest,
+      });
+    }
+```
+
+#### `entries`
+**Type**: `() => Array<[id: string, DataEntry]>`
```

Comment:
I'm just double-checking since this seems mostly inspired by JS' `Map`. Is this an actual array, or is it an iterator yielding those tuples?

Reply:
It's an actual array